handlers/UpdateEmployee.go

```diff
@@ -87,20 +74,63 @@ func UpdateEmployee(w http.ResponseWriter, req *http.Request) {
 		Department: updatedEmployee.Department,
 	}
 
-	//print out old employee info!
+	//Old employee info!
 	fmt.Println("BEFORE UPDATING INFO: ", currentEmployeeInfo)
-
+	//Patched employee info
 	fmt.Println("AFTER UPDATING INFO: ", newEmployeeCopy)
 
-	store.Employees[validID] = newEmployeeCopy
-
-	w.WriteHeader(http.StatusOK)
-	w.Header().Set("Content-Type", "application/json")
+	//check all struct fields
+	if newEmployeeCopy.Name == "" {
+		newEmployeeCopy.Name = currentEmployeeInfo.Name
+	}
+	if newEmployeeCopy.Email == "" {
+		newEmployeeCopy.Email = currentEmployeeInfo.Email
+	}
+	if newEmployeeCopy.EmployeeId == "" {
+		newEmployeeCopy.EmployeeId = currentEmployeeInfo.EmployeeId
+	}
+	if newEmployeeCopy.City == "" {
+		newEmployeeCopy.City = currentEmployeeInfo.City
+	}
+	if newEmployeeCopy.Address == "" {
+		newEmployeeCopy.Address = currentEmployeeInfo.Address
+	}
+	if newEmployeeCopy.Department == "" {
+		newEmployeeCopy.Department = currentEmployeeInfo.Department
+	}
```

Comment:
I also considered using a more dynamic approach with ->
`———————————————————
v := reflect.ValueOf(newEmployeeCopy)
typeOfEmp := v.Type()

v2 := reflect.ValueOf(currentEmployeeInfo)
//typeOfEmp2 := v.Type()

for i := 0; i < v.NumField(); i++ {
	fmt.Printf("Field: %s\tValue: %v\n", typeOfEmp.Field(i).Name, v.Field(i).Interface())
	//field := typeOfEmp.Field(i).Name
	value := v.Field(i).Interface()

	if value == "" {
		value = v2.Field(i).Interface()
	}
}

fmt.Println(v2)
———————————————————`

Reply:
However, i ultimately decided against it in this case due to legibility of code. I think in the case where I need to assure many struct fields need to be populated, and I don't know which fields they are then the approach above may merit consideration. Definitely open to hearing pros/cons of either approach though!

Reply:
The variadic options pattern may have worked here too, potentially, but nothing wrong with the way you have it.